### PR TITLE
unet_2nd_pull_46271826

### PR DESCRIPTION
**Pull request by FrostNov4**
A pull request was made on the main branch last time, this one is made on the recognition branch, which contains the relevant files.

Code completed 27 OCT,

Had an extension to 27 OCT.

The code is free of bugs and got a test dice score of 99%

### CODE REVIEW

**Comment by wangzhaomxy** [comment]
- Wrong file path. All student-constructed documents should be packed in one document under the ‘recognition’ document.
- Algorithm solves the problem appropriately. Claimed Average Dice Coefficient in the testing dataset: 0.99, but this doesn’t make sense. Predicted Mask output looks good. Containing data augmentation code but has been commented off in Data_loader.py, so no data augmentation. Training and validating are in the same whole dataset, data leakage, and this is also the reason why the model got a high dice coefficient. Other parts are ok.
- Using Pytorch to construct the Improved U-Net models and functions. However, the model body is roughly an improved U-Net model(filters start with 16x4 and end up with 128x4) instead of the improved U-Net in the paper(filters start with 16 and end up with 256). The active function(sigmoid) at the end of the output has been commented off. There’s only one function in predict.py, the suspected implementation part has been commented off, so cannot use the predict.py to generate anything. In addition, the input and output variable names are not defined in the file. Other parts are ok.
- Good coding design. 
- Sufficient docstring and comment in the code, good references. 
- Easy difficulty problem, no higher difficulty marks.

**Comment by shakes76** [comment]
# Marking
## Good Practice (Design/Commenting, TF/Torch Usage)
Adequate design and implementation, but not Improved UNet, only UNet -4
Good spacing and comments
Header blocks
## Recognition Problem 
Solves problem but no working prediction -2
Driver Script present
File structure present
Shows Usage & Demo & Visualisation & Data usage
Module present
Commenting
Data leakage present -1, no augmentation -1
Difficulty: Easy -10
## Commit Log
Meaningful commit messages minimal -1
Progressive commits used
## Documentation
ReadMe OK, no usage -1, no refs -1
Model/technical explanation minimal -1
Good Description and Comments 
Markdown used and PDF submitted
## Pull Request
Successful Pull Request (Working Algorithm Delivered on Time in Correct Branch) 
Feedback required, remove .idea folder and files, restore repo READMEs -2
Request Description minimal -1

**Comment by shakes76** [comment]
Feedback marks possible +2 if the requested changes are made (see above).

**Comment by wangzhaomxy** [comment]
No feedback attempt and no feedback marks granted.

**Comment by FrostNov4** [comment]
Hi there,

I am wondering in order to gain the 2 marks,  am I supposed to fix all of the mistakes or just about the readMe or any parts of my report?

Thank you in advance!

**Comment by wangzhaomxy** [comment]
> Hi there,
> 
> I am wondering in order to gain the 2 marks, am I supposed to fix all of the mistakes or just about the readMe or any parts of my report?
> 
> Thank you in advance!

"Feedback required, remove .idea folder and files, restore repo READMEs -2"

Just following Shakes' instruction to modify your files by the end of today. I'll add 2 marks after you do this. The modification of other parts is not necessary.

**Comment by FrostNov4** [comment]
modification according to the feedback:
- model modified to follow the same structure of improved Unet in the paper,
- data normalization and augmentation were added for data preprocessing,
- solving data leakage by splitting the training data into a training set and a validation set.
- prediction.py can be run to get predictions shown in readme directly.
- code usage examples were added to the readme file.
- more explanations were added to the readme file.
- references were added to the readme file.
- image can be shown properly in the readme file.

Besides, although I haven't retrained the model for the modified model and data preprocessing steps, the code should be free of bugs as I have ran several demos before pushing the commit onto Github.

**Comment by wangzhaomxy** [comment]
Hard work on the feedback attempt, still having some file problems. Please see Shakes' inspection and try to modify fix them. But I can give you feedback marks granted +2.

**Comment by FrostNov4** [comment]
OK Thank you for the +2 feedback marks! 

I have tried to fix the file problems. However, too many commits are associated with the folder outside the recognition folder, as I initially implemented my project outside the recognition file. I have no idea how I can remove these changes without affecting my codes.
